benchmarks/str-locality.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+
+import time
+import argparse
+import arkouda as ak
+
+def generate_data(N, seed):
+    prefix = ak.random_strings_uniform(minlen=1, maxlen=16, size=N, seed=seed, characters='numeric')
+    if seed is not None:
+        seed += 1
+    suffix = ak.random_strings_uniform(minlen=1, maxlen=16, size=N, seed=seed, characters='numeric')
+    random_strings = prefix.stick(suffix, delimiter='.')
+    perm = ak.argsort(random_strings.get_lengths())
+    sorted_strings = random_strings[perm]
+    return random_strings, sorted_strings, perm
+
+def time_all_ops(N, trials, seed, correctnessOnly):
+    cfg = ak.get_config()
+    if correctnessOnly:
+        N = 10**4
+    else:
+        print(">>> arkouda substring search")
```

Comment:
Not just `substring search`. Maybe `string locality tests`?

Reply:
good catch -- serves me right for copy/pasting :man_facepalming: